src/GafferScene/FramingConstraint.cpp

```diff
@@ -0,0 +1,687 @@
+	result->members()[1] = const_cast< IECore::Object *>( object );
+
+	static_cast<ObjectVectorPlug *>( output )->setValue( result );
+}
+
+} // namespace
+
+void FramingConstraint::hash( const Gaffer::ValuePlug *output, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	SceneElementProcessor::hash( output, context, h );
+	if( output != transformAndObjectPlug() )
+	{
+		return;
+	}
+
+	const ScenePath &path = context->get<ScenePath>( ScenePlug::scenePathContextName );
+	const ScenePath parentPath( path.begin(), path.end() - 1 );
+	inPlug()->transformPlug()->hash( h );
+	h.append( inPlug()->fullTransformHash( parentPath ) );
+	inPlug()->objectPlug()->hash( h );
+
+	auto targetOpt = target();
+	if( !targetOpt )
+	{
+		return;
+	}
+
+	boundModePlug()->hash( h );
+	extendFarClipPlug()->hash( h );
+	paddingPlug()->hash( h );
+
+	std::unique_ptr<Context::EditableScope> useFrameScope;
+	if( useTargetFramePlug()->getValue() )
+	{
+		useFrameScope = std::make_unique< Context::EditableScope >( context );
+		useFrameScope->setFrame( targetFramePlug()->getValue() );
+	}
+
+	h.append( targetOpt->scene->fullTransformHash( targetOpt->path ) );
+	h.append( targetOpt->scene->boundHash( targetOpt->path ) );
+	h.append( targetOpt->scene->childBoundsHash( targetOpt->path ) );
+}
+
+void FramingConstraint::compute( Gaffer::ValuePlug *output, const Gaffer::Context *context ) const
+{
+	if( output != transformAndObjectPlug() )
+	{
+		SceneElementProcessor::compute( output, context);
+		return;
+	}
+
+	// Read the local transform and full transform of the parent separately and combine them to find
+	// the world camera transform, because the final camera transform we output needs to be relative
+	// to the parent
+	M44f cameraTransform = inPlug()->transformPlug()->getValue();
+	const ScenePath &path = context->get<ScenePath>( ScenePlug::scenePathContextName );
+	const ScenePath parentPath( path.begin(), path.end() - 1 );
+	M44f parentTransform = inPlug()->fullTransform( parentPath );
+	M44f worldCameraTransform = cameraTransform * parentTransform;
+
+	ConstObjectPtr object = inPlug()->objectPlug()->getValue();
+	IECoreScene::ConstCameraPtr camera = IECore::runTimeCast< const IECoreScene::Camera >( object.get() );
+
+	auto targetOpt = target();
+	if( !targetOpt || !camera )
+	{
+		setTransformAndObject( output, cameraTransform, object.get() );
+		return;
+	}
+
+	M44f targetTransform;
+	Box3f targetBound;
+
+	{
+		std::unique_ptr<Context::EditableScope> useFrameScope;
+		if( useTargetFramePlug()->getValue() )
+		{
+			useFrameScope = std::make_unique< Context::EditableScope >( context );
+			useFrameScope->setFrame( targetFramePlug()->getValue() );
+		}
+
+		targetTransform = targetOpt->scene->fullTransform( targetOpt->path );
+		targetBound.extendBy( targetOpt->scene->bound( targetOpt->path ) );
+		targetBound.extendBy( targetOpt->scene->childBounds( targetOpt->path ) );
```

Comment:
Why are we separately getting the child bounds here too? `bound()` should already includes `childBounds()`.